test/apps/iodemo/run_io_demo.sh

```diff
@@ -0,0 +1,456 @@
+#!/bin/sh
+#
+# Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+#
+# See file LICENSE for terms.
+#
+
+# set $? to zero iff verbose mode is on
+is_verbose()
+{
+	[ ${verbose} -ne 0 ]
+}
+
+# Show a message in verbose mode only
+verbose_log()
+{
+	is_verbose && echo "$@"
+}
+
+# Show the value of a variable
+show_var()
+{
+	value=$(eval echo \$$1)
+	printf "    %20s : '%s'\n" $1 "${value}"
+}
+
+# Show the value of a variable only if verbose
+show_var_verbose()
+{
+	is_verbose && show_var $1
+}
+
+# Split a comm-separated list to space-separated list which can be used by 'for'
+split_list()
+{
+	echo $1 | tr ',' ' '
+}
+
+# return minimal number of two integer arguments
+min()
+{
+	[ $1 -lt $2 ] && echo $1 || echo $2
+}
+
+# Update some default configuration settings according to slurm launch
+check_slurm_env()
+{
+	if [ -z "$SLURM_JOBID" ]
+	then
+		# Skip slurm
+		return
+	fi
+
+	# Nodes to run on
+	host_list=$(hostlist -s, -e $(squeue -j ${SLURM_JOBID} -h -o "%N"))
+	SLURM_NNODES=$(squeue -j ${SLURM_JOBID} -h -o "%D")
+	NNODES=$SLURM_NNODES
+
+	if [ -n "$SLURM_TASKS_PER_NODE" ]
+	then
+		tasks_per_node=$(echo $SLURM_TASKS_PER_NODE | cut -d'(' -f1)
+	elif [ -n "$SLURM_JOB_CPUS_PER_NODE" ]
+	then
+		tasks_per_node=$(echo $SLURM_JOB_CPUS_PER_NODE | cut -d'(' -f1)
+	else
+		TOTAL_CPUS=$(squeue -j ${SLURM_JOBID} -h -o "%C")
+		tasks_per_node=$((${TOTAL_CPUS} / ${SLURM_NNODES}))
+	fi
+}
+
+# Set default configuration values
+init_config()
+{
+	verbose=0
+	iodemo_exe=""
+	iodemo_client_args=""
+	num_clients=1
+	num_servers=1
+	map_by="node"
+	base_port_num=20000
+	tasks_per_node=1
+	net_if="bond0"
+	duration=30
+
+	# command line args will override slurm env vars
+	check_slurm_env
+}
+
+# Show the configuration
+show_config()
+{
+	echo "Launch configuration:"
+	for key in \
+			host_list tasks_per_node map_by \
+			num_clients num_servers \
+			iodemo_exe iodemo_client_args \
+			net_if base_port_num \
+			duration
+	do
+		show_var ${key}
+	done
+}
+
+# Helper to show default value if defined, otherwise empty string
+show_default_value()
+{
+	value=$(eval echo \$$1)
+	[ -z "${value}" ] && echo "" || echo " (${value})"
+}
+
+# Show script usage help message
+usage()
+{
+	echo "Usage: run_io_demo.sh <options> <exe> <arguments>"
+	echo
+	echo "  -h|--help                   Show this help message"
+	echo "  -v|--verbose                Turn on verbosity"
+	echo "  -H|--hostlist <h1>,<h2>,..  List of host names to run on"$(show_default_value host_list)
+	echo "  -i|--netif <n>              Network interface to use"$(show_default_value net_if)
```

Comment:
btw, shall we have bond i/f in testplan? also multi-rail?

Reply:
the network interface is used as the IP to connect to, default valud is bond0